shared/src/test/scala/org/adridadou/openlaw/vm/OpenlawVmSpec.scala

```diff
@@ -87,6 +87,40 @@ class OpenlawVmSpec extends FlatSpec with Matchers {
     vm1.signature(identity2.email) shouldBe None
   }
 
+  it should "be possible to sign the contract with an external signature" in {
+    val template = """this is a contract [[Signatory:ExternalSignature(service:"DocuSign")]]"""
+    val Right(identity: Identity) = Email("hello@world.com").map(Identity.withEmail)
+    val templateId = TemplateId(TestCryptoService.sha256(template))
+
+    val serviceAccount = TestAccount.newRandom
+    val serviceName = ServiceName("DocuSign")
+
+    val definition = ContractDefinition(
+      creatorId = UserId.SYSTEM_ID,
+      mainTemplate = templateId,
+      templates = Map(),
+      parameters = TemplateParameters(Map(VariableName("Signatory") -> ExternalSignatureType.internalFormat(ExternalSignature(serviceName = serviceName, identity = Some(identity))).right.value))
+    )
+
+    val vm = vmProvider.create(definition, None, OpenlawSignatureOracle(TestCryptoService, serverAccount.address, Map(serviceName -> serviceAccount.address)), Seq())
+
+    vm(LoadTemplate(template))
+    vm.executionState shouldBe ContractCreated
+    vm.allNextActions.map(_.map(_.action)) shouldBe Right(Seq(SignatureAction(identity.email)))
+
+    val badSignature = EthereumSignature(signByEmail(identity.email, vm.contractId.data, serverAccount).signature)
+    val signature = EthereumSignature(signByEmail(identity.email, vm.contractId.data, serviceAccount).signature)
```

Comment:
What is the difference between the `badSignature` and `signature`?

Reply:
it's the private key used to generate the signature. The idea is to show that if you try to sign with the wrong private key, it doesn't work

Reply:
if you look closely, one is using serverAccount where the other uses serviceAccount

Reply:
Got it. Thanks!